packages/next/src/utils.ts

```diff
@@ -2666,3 +2668,50 @@ function transformSourceMap(
 
   return { ...sourcemap, sources };
 }
+
+interface LambdaGroupTypeInterface {
+  isApiLambda: boolean;
+  isPrerenders?: boolean;
+}
+
+export function getOperationType({
+  group,
+  prerenderManifest,
+  pageFileName,
+}: {
+  group?: LambdaGroupTypeInterface;
+  prerenderManifest?: NextPrerenderedRoutes;
+  pageFileName?: string;
+}) {
+  if (group?.isApiLambda || isApiPage(pageFileName)) {
+    return 'API';
+  }
+
+  if (group?.isPrerenders) {
+    return 'ISR';
+  }
+
+  if (pageFileName) {
+    const routePaths = Object.keys({
+      ...prerenderManifest?.staticRoutes,
+      ...prerenderManifest?.blockingFallbackRoutes,
+      ...prerenderManifest?.fallbackRoutes,
+    });
+
+    if (routePaths.includes(pageFileName)) {
+      return 'ISR';
+    }
+  }
```

Comment:
Is this the right way to check the `prerenderManifest`?

Reply:
Only `blockingFallbackRoutes` or `fallbackRoutes` would match `pageFileName` here static routes are the normal path not the page e.g. `/blog/first` instead of `/blog/[slug]`